File: DynamicSunTest/DynamicSunTest/Models/WeatherData.cs
namespace DynamicSunTest.Models
{
    public sealed class WeatherData
    {
        public DateTime Date { get; set; }

        public DateTime Time { get; set; }

        public double Temperature { get; set; }

        public int Humidity { get; set; }

        public double DewPoint { get; set; }

        public int AtmospherePressure { get; set; }

        public string WindDirection { get; set; }

        public int? WindSpeed { get; set; }

        public int? Cloudy { get; set; }

        public int? CloudBase { get; set; }

        public string WeatherConditions { get; set; }
    }
}

File: DynamicSunTest/DynamicSunTest/wwwroot/js/weatherData.js
$(document).ready(function () {
    const table = $('#weatherTable').DataTable({
        paging: true,
        pageLength: 50,
    });

    // Получение года из параметров URL
    const urlParams = new URLSearchParams(window.location.search);
    const year = urlParams.get('year');

    // Вызов метода бэкенда для получения данных
    fetch(`/WeatherView/GetWeatherData?year=${year}`)
        .then(response => response.json())
        .then(data => {
            table.clear();

            data.forEach(item => {
                table.row.add([
                    item.date,
                    item.time,
                    item.temperature,
                    item.humidity,
                    item.dewPoint,
                    item.atmospherePressure,
                    item.windDirection,
                    item.windSpeed,
                    item.cloudy,
                    item.cloudBase,
                    item.weatherConditions
                    // Добавьте другие значения, если необходимо
                ]);
            });

            table.draw();
        });

});